public/app/features/variables/editor/selectors.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  AdHocVariableEditorState,
+  DataSourceVariableEditorState,
+  QueryVariableEditorState,
+  VariableEditorState,
+} from './reducer';
+
+/**
+ * Narrows generic variable editor state down to specific Adhoc variable extended editor state
+ */
+export function getAdhocVariableState(editorState: VariableEditorState): AdHocVariableEditorState | null {
```

Comment:
super small nit: can we name these selectors `get<Type>EditorState` instead?

Reply:
Totally - I haven't thought too much about naming in this at all, and was hoping you would pick some things up!